client/lib/react-query-devtools-helper/index.jsx

```diff
@@ -0,0 +1,55 @@
+import config from '@automattic/calypso-config';
+import { useEffect, useState } from 'react';
+import ReactDom from 'react-dom';
+import { ReactQueryDevtools } from 'react-query/devtools';
+import { setStoredItem, getStoredItem } from 'calypso/lib/browser-storage';
+
+import './style.scss';
+
+function useDevtoolsEnabled() {
+	const [ enableDevtools, setEnableDevtools ] = useState();
+
+	useEffect( () => {
+		getStoredItem( 'enable-react-query-devtools' ).then( setEnableDevtools );
+	}, [] );
+
+	return enableDevtools;
+}
+
+export function CalypsoReactQueryDevtools() {
+	const enableDevtools = useDevtoolsEnabled();
+
+	if ( config.isEnabled( 'dev/react-query-devtools' ) ) {
+		return enableDevtools ? <ReactQueryDevtools /> : null;
+	}
+
+	return null;
+}
+
+function ReactQueryDevtoolsHelper() {
+	const enableDevtools = useDevtoolsEnabled();
+
+	async function toggleDevtools( event ) {
+		await setStoredItem( 'enable-react-query-devtools', event.target.checked );
```

Comment:
It would be nice if the `'enable-react-query-devtools'` stored item was encapsulated at one place. To achieve that, `useDevtoolsEnabled` could have a `useState`-ish API:
```js
const [ devtoolsEnabled, setDevtoolsEnabled ] = useDevtoolsEnabled();

const toggle = () => {
  await setDevtoolsEnabled( event.target.checked );
};
```

Reply:
Great suggestion. Implemented via https://github.com/Automattic/wp-calypso/pull/52058/commits/095dcc49b5b4f030e663ff20b1cd51cf8b8df3d9 👍🏻 let me know if that's what you had in mind.